### ISSUE TITLE
optparse.OptionConflictError: option -v/--verbosity w/ Django 1.8

### ISSUE THREAD

**Issue by paulj3000** [gabrielfalcao/lettuce#480]
Hi,

I recently upgraded to Django 1.8 and lettuce is throwing the following error:

optparse.OptionConflictError: option -v/--verbosity: conflicting option string(s): -v, --verbosity

I found the error in this file: lettuce/django/management/commands/harvest.py
on line 40..
make_option('-v', '--verbosity', action='store', dest='verbosity', default='4',

If I change to 
make_option('-j', '--jverbosity', action='store', dest='jverbosity', default='4',

and line 127 to
verbosity = int(options.get('jverbosity', 4))

everything works great

Thanks

**Comment by chris-erickson** [comment]
I'd love to see a fix for 1.8.  I was hoping to start making use of this in a new project.

**Comment by paul-butcher** [comment]
The problem starts with django/django@856863860352aa1f0288e6c9168a0e423c4f5184
There is no longer a prepopulated option_list to append to and/or override.  The built-in options are added separately.

On line 39 of lettuce/django/management/commands/harvest.py, it attempts to override the built-in '-v', by discarding the first default option (which was '-v'), thus: `option_list = BaseCommand.option_list[1:] + (`.

This no longer overrides anything.

**Comment by hmleal** [comment]
In older versions of django option_list returns:

(<Option at 0x7fbc36e96ab8: -v/--verbosity>,
 <Option at 0x7fbc36e96320: --settings>,
 <Option at 0x7fbc36e96908: --pythonpath>,
 <Option at 0x7fbc36e96f38: --traceback>)

Now is a empty tuple.

**Comment by hmleal** [comment]
Here are the detailed description of what has changed
https://docs.djangoproject.com/en/1.8/howto/custom-management-commands/

**Comment by hmleal** [comment]
I think the best way out now is to rename the --verbosity to --lettuce-verbosity to avoid conflict.

**Comment by claudep** [comment]
A possible workaround:
 * remove the `make_option('-v', '--verbosity'...`
 * override the `create_parser` method, call the parent, call `parser.remove_option('-v')` followed by parser.add_option(...)` with your verbosity definition.

**Comment by michelts** [comment]
@giang271291 and @hmleal, I think the solution proposed by @claudep would be better. Create a new lettuce-verbosity argument would left a verbosity argument unused, I think it is a waste.

If we go further in the problem, lettuce API could be improved to use only the [default options for verbosity](https://docs.djangoproject.com/en/1.8/ref/django-admin/#django-admin-option---verbosity), between 0 and 3. Today, we have an extra 4 option for colorized output (the default option).

We could replace the 4th option by a combination of verbosity 3 and the [no-color argument](https://docs.djangoproject.com/en/1.8/ref/django-admin/#django-admin-option---no-color).

I would run harvest as follow:

```
python manage.py harvest -v 3  # normal colorized output
python manage.py harvest -v 3 --no-color  # colorless output
python manage.py harvest -v 2  # only scenario names
python manage.py harvest -v 1  # only dots
python manage.py harvest -v 0  # no output
```

The --no-color argument was added in django 1.7 but I'm sure there is a way to check if it is available, we could test the django version or the --no-color presence to add the 4th option to verbosity.

We still need to replace the default django verbosity to customize his help_text and default value, but it would be great to move the same direction as django. @gabrielfalcao and @danni, what do you think about this approach?

**Comment by paul-butcher** [comment]
Is there a reason for wanting to change this whilst continuing to use the deprecated optparse mechanism?

If a change is required, then it should probably be updated to use the argparse instead.

**Comment by claudep** [comment]
I guess that it depends on the Django compatibility policy of this package. If Django < 1.8 doesn't need to be supported any more, then of course, the migration to argparse is the way to go.

**Comment by michelts** [comment]
We should support at least django 1.4 LTS and django 1.7 until october, as seem in the supported versions section of the [django download page](https://www.djangoproject.com/download/).

**Comment by hmleal** [comment]
In Lettuce the minimum Django version is 1.5.5, but 1.5 and 1.6 are not maintained anymore. So i think the minimum Django version supported by lettuce should be 1.7 so we could remove the -v 4 option and just use the -v 3 --no-color.

**Comment by ykhrustalev** [comment]
guys, sorry for asking here, but is there a alive fork of that repo, where pull requests get merged?

**Comment by paulj3000** [comment]
I made a quick fix for this issue, you can find it here:

https://github.com/paulj3000/lettuce

Note, this is very quick until an official release can be made.  I also fixed a deprecation warning.  This is good w/ >=python 2.7.6


Good luck!

**Comment by alu0100837094** [comment]
paulj3000 you just saved my college project :D , a weeks ago i saw your post and it works :trophy: @paulj3000

**Comment by paulj3000** [comment]
My pleasure!

**Comment by michelts** [comment]
I think some manager can can mark this issue as done, once the pull request is merged...

**Comment by michelts** [comment]
Thansk @gabrielfalcao ;)

**Comment by sufendyc** [comment]
Hi, looks like the issue is fixed but not released yet. How do I get this fix if I need to install lettuce using pip?

**Comment by michelts** [comment]
We use git+git://github.com/gabrielfalcao/lettuce in our requirements.txt file.

**Comment by gacpro** [comment]
Hi, I'm in charge of a web site development and it will be released in a few weeks. We need this fix to be released within a new Django version ASAP please! How can I contact Django team to get this done?

**Comment by gacpro** [comment]
I've added git+git://github.com/gabrielfalcao/lettuce to my requirements.txt file, but the issue is still happening.

**Comment by danni** [comment]
It'll be something like `-e git://foo#egg=lettuce-dev`

Alternatively consider switching to Aloe (github.com/koterpillar/aloe) which is the successor of the Lettuce Infoxchange/new-parser branch.

**Comment by michelts** [comment]
You don't need to turn it editable, see my requirements.txt file:

```
git+git://github.com/gabrielfalcao/lettuce
```

If i run a `pip install -r requirements.txt`, I will get the output:

```
Downloading/unpacking git+git://github.com/gabrielfalcao/lettuce (from -r requirements.txt (line 1))
  Cloning git://github.com/gabrielfalcao/lettuce to ./pip-4mN0FX-build
  Running setup.py (path:/tmp/pip-4mN0FX-build/setup.py) egg_info for package from git+git://github.com/gabrielfalcao/lettuce
...

Successfully installed sure fuzzywuzzy python-subunit lettuce mock six extras testtools funcsigs pbr traceback2 unittest2 python-mimeparse linecache2
Cleaning up...
```

You may check if pip *updated* your local lettuce installation, maybe it refuse to replace the copy installed after a `pip install lettuce`. Try rebuild the virtual environment.

**Comment by gabrielfalcao** [comment]
just released the version 0.2.21
